src/main/java/org/skriptlang/skript/bukkit/spark/SparkUtils.java

```diff
@@ -0,0 +1,162 @@
+	private static volatile DoubleStatistic<CpuUsage> cpuSystem;
+	private static volatile GenericStatistic<DoubleAverageInfo, MillisPerTick> mspt;
+	private static volatile DoubleStatistic<TicksPerSecond> tps;
+
+	@Contract(pure = true)
+	private static @NotNull Spark spark() {
+		return SparkProvider.get();
+	}
+
+	/**
+	 * Checks for MSPT support. This should be checked before using the MSPT methods.
+	 * @return whether MSPT statistics are supported.
+	 */
+	public static boolean supportsMspt() {
+		return mspt() != null;
+	}
+
+	private static @UnknownNullability GenericStatistic<DoubleAverageInfo, MillisPerTick> mspt() {
+		if (mspt == null) {
+			synchronized (SparkUtils.class) {
+				if (mspt == null)
+					mspt = spark().mspt();
+			}
+		}
+		return mspt;
+	}
+
+	/**
+	 * Get the MSPT stat info block for the given window.
+	 * {@link #supportsMspt()} should be checked prior to using this.
+	 *
+	 * @param window the window for which to gather statistics from.
+	 * @return the MSPT info block.
+	 */
+	public static @Nullable DoubleAverageInfo mspt(MillisPerTick window) {
+		GenericStatistic<DoubleAverageInfo, MillisPerTick> mspt = mspt();
+		return mspt.poll(window);
+	}
+
+	/**
+	 * Get the MSPT stat info block for the given windows.
+	 * {@link #supportsMspt()} should be checked prior to using this.
+	 *
+	 * @param windows the windows for which to gather statistics from.
+	 * @return the MSPT info blocks.
+	 */
+	public static DoubleAverageInfo @Nullable [] mspt(MillisPerTick... windows) {
+		GenericStatistic<DoubleAverageInfo, MillisPerTick> mspt = mspt();
+		return Arrays.stream(windows).map(mspt::poll).toArray(DoubleAverageInfo[]::new);
+	}
+
+	private static DoubleStatistic<CpuUsage> cpuProcess() {
+		if (cpuProcess == null) {
+			synchronized (SparkUtils.class) {
+				if (cpuProcess == null)
+					cpuProcess = spark().cpuProcess();
+			}
+		}
+		return cpuProcess;
+	}
+
+	/**
+	 * Get the process CPU usage for the given window.
+	 * @param window the window for which to gather statistics from.
+	 * @return the process cpu usage.
+	 */
+	public static double cpuProcess(CpuUsage window) {
+		return cpuProcess().poll(window);
+	}
+
+	/**
+	 * Get the process CPU usage for the given windows.
+	 * @param windows the windows for which to gather statistics from.
+	 * @return the process cpu usages.
+	 */
+	public static Double @NotNull [] cpuProcess(CpuUsage... windows) {
+		DoubleStatistic<CpuUsage> cpu = cpuProcess();
+		return Arrays.stream(windows).map(cpu::poll).toArray(Double[]::new);
+	}
+
+	private static DoubleStatistic<CpuUsage> cpuSystem() {
+		if (cpuSystem == null) {
+			synchronized (SparkUtils.class) {
+				if (cpuSystem == null)
+					cpuSystem = spark().cpuSystem();
+			}
+		}
+		return cpuSystem;
+	}
+
+	/**
+	 * Get the system CPU usage for the given window.
+	 * @param window the window for which to gather statistics from.
+	 * @return the system cpu usage.
+	 */
+	public static double cpuSystem(CpuUsage window) {
```

Comment:
`pollSystemCpuUsage(window)`
When it's named like this, I no longer need to rely on one-liner documentation. This also makes call sites more intuitive.